packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -210,14 +210,15 @@ class LocalEngineArtifacts extends Artifacts {
     return fs.path.basename(engineOutPath);
   }
 
-  String _genSnapshotPath(TargetPlatform platform, BuildMode mode) {
-    String clang;
-    if (platform == TargetPlatform.ios) {
-      clang = 'clang_x64';
-    } else {
-      clang = getCurrentHostPlatform() == HostPlatform.darwin_x64 ? 'clang_i386' : 'clang_x86';
+  String _genSnapshotPath() {
+    const List<String> clangDirs = const <String>['clang_x86', 'clang_x64', 'clang_i386'];
+    final String genSnapshotName = _artifactToFileName(Artifact.genSnapshot);
+    for (String clangDir in clangDirs) {
+      final String genSnapshotPath = fs.path.join(engineOutPath, clangDir, genSnapshotName);
+      if (fs.file(genSnapshotPath).existsSync())
+        return genSnapshotPath;
     }
-    return fs.path.join(engineOutPath, clang, _artifactToFileName(Artifact.genSnapshot));
+    throw new Exception('Unable to find $genSnapshotName');
```

Comment:
We'd want to make sure that this results in a tool crash with a crash report. Could you please check that that's what happens?

Also, do we expect that this line should never be reached? If so, we should throw a longer message saying that this is likely a bug in Flutter and invite the user to file an issue. [Here's an example](https://github.com/flutter/flutter/blob/60f5ae8e77ecfd479333ce68a8fd81b0778132f1/packages/flutter/lib/src/foundation/assertions.dart#L277).